pkg/controller/template/render.go

```diff
@@ -462,3 +465,37 @@ func onPremPlatform(platformString configv1.PlatformType) bool {
 		return false
 	}
 }
+
+// urlHost is a template function that returns the hostname of a url (without the port)
+func urlHost(u string) (interface{}, error) {
+	parsed, err := url.Parse(u)
+	if err != nil {
+		return nil, fmt.Errorf("invalid url: %w", err)
+	}
+
+	return parsed.Hostname(), nil
+}
+
+// urlPort is a template function that returns the port of a url, with defaults
+// provided if necessary.
+func urlPort(u string) (interface{}, error) {
+	parsed, err := url.Parse(u)
+	if err != nil {
+		return nil, fmt.Errorf("invalid url: %w", err)
+	}
+
+	port := parsed.Port()
+	if port != "" {
+		return port, nil
+	}
+
+	// default port
+	switch parsed.Scheme {
+	case "https":
+		return "443", nil
+	case "http":
+		return "80", nil
+	default:
```

Comment:
What would happen if this were to fall to any non-standard port? Is there a way to dynamically get this?

Reply:
yeah, the above line handles this case (since 6443 is, of course, non-standard)

Reply:
so it would return unknown scheme for 6443? am I missing something around here?

Reply:
you've got it backwards - what this code does is ensure `https://foo.com` returns port 443, but `https://foo.com:1234` returns 1234.